e2e/cypress/integration/account_settings/general/username_spec.js

```diff
@@ -0,0 +1,203 @@
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @account_setting
+
+describe('Account Settings > Sidebar > General > Edit', () => {
+    let testTeam;
+    let testChannel;
+    let otherUser;
+
+    before(() => {
+        // # Login as admin and visit town-square
+        cy.apiInitSetup().then(({team, channel}) => {
+            testTeam = team;
+            testChannel = channel;
+
+            cy.apiCreateUser({prefix: 'other'}).then(({user: user1}) => {
+                otherUser = user1;
+
+                cy.apiAddUserToTeam(testTeam.id, otherUser.id).then(() => {
+                    cy.apiAddUserToChannel(testChannel.id, otherUser.id);
+                });
+            });
+
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+    });
+
+    beforeEach(() => {
+        // # Go to Account Settings
+        cy.toAccountSettingsModal();
+    });
+
+    it('MM-T2050 Username cannot be blank', () => {
+        // # Click the General tab
+        cy.get('#generalButton').should('be.visible').click();
+
+        // # Clear the username textfield contents
+        cy.get('#usernameEdit').click();
+        cy.get('#username').clear();
+        cy.get('#saveSetting').click();
+
+        // * Check if element is present and contains expected text values
+        cy.get('#clientError').should('be.visible').should('contain', 'Username must begin with a letter, and contain between 3 to 22 lowercase characters made up of numbers, letters, and the symbols \'.\', \'-\', and \'_\'.');
+
+        // # Click "x" button to close Account Settings modal
+        cy.get('#accountSettingsHeader > .close').click();
+    });
+
+    it('MM-T2051 Username min 3 characters', () => {
+        // # Click the General tab
+        cy.get('#generalButton').should('be.visible').click();
+
+        // # Edit the username field
+        cy.get('#usernameEdit').click();
+
+        // # Add the username to textfield contents
+        cy.get('#username').clear().type('te');
+        cy.get('#saveSetting').click();
+
+        // * Check if element is present and contains expected text values
+        cy.get('#clientError').should('be.visible').should('contain', 'Username must begin with a letter, and contain between 3 to 22 lowercase characters made up of numbers, letters, and the symbols \'.\', \'-\', and \'_\'.');
+
+        // # Click "x" button to close Account Settings modal
+        cy.get('#accountSettingsHeader > .close').click();
+    });
+
+    //Note: the test case mentions an error message that could not be found in the codebase: "This username is already taken. Please choose another." Added a note to the testcase to update the error message.
```

Comment:
Thanks for checking! Looks like it was recently changed by https://github.com/mattermost/mattermost-server/pull/15563. I've update the test case to reflect the new error message: "Unable to find the existing account to update.".
```suggestion
```